### PR TITLE
Revert API breakage in IStorage

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
https://github.com/owncloud/core/pull/32051 was possible a bit too aggressive ...

## Motivation and Context
We shall not break APIs for apps

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: lib/public/Files/Storage.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * Provide a common interface to all different storage options
+ *
+ * All paths passed to the storage are relative to the storage and should NOT have a leading slash.
+ *
+ * @since 6.0.0
+ * @deprecated 9.0.0 use \OCP\Files\Storage\IStorage instead
+ */
+interface Storage extends IStorage {
```
Comment (PVince81) [inline lib/public/Files/Storage.php]: would it work to only bring back this storage interface, but keep using IStorage everywhere ?

Comment (DeepDiver1975) [reply lib/public/Files/Storage.php]: Unfortunatly not

```php
[Tue Jul 24 13:03:18 2018] PHP Fatal error:  Declaration of OCA\Ransomware_Protection\StorageWrapper::copyFromStorage(OCP\Files\Storage $sourceStorage, $sourceInternalPath, $targetInternalPath) must be compatible with OCP\Files\Storage\IStorage::copyFromStorage(OCP\Files\Storage\IStorage $sourceStorage, $sourceInternalPath, $targetInternalPath) in /var/www/owncloud/apps/ransomware_protection/lib/StorageWrapper.php on line 0
[Tue Jul 24 13:03:18 2018] 127.0.0.1:36352 [500]: /ocs/v1.php/apps/testing/api/v1/increasefileid - Declaration of OCA\Ransomware_Protection\StorageWrapper::copyFromStorage(OCP\Files\Storage $sourceStorage, $sourceInternalPath, $targetInternalPath) must be compatible with OCP\Files\Storage\IStorage::copyFromStorage(OCP\Files\Storage\IStorage $sourceStorage, $sourceInternalPath, $targetInternalPath) in /var/www/owncloud/apps/ransomware_protection/lib/StorageWrapper.php on line 0
[Tue Jul 24 13:03:18 2018] Declaration of OCA\Ransomware_Protection\StorageWrapper::copyFromStorage(OCP\Files\Storage $sourceStorage, $sourceInternalPath, $targetInternalPath) must be compatible with OCP\Files\Storage\IStorage::copyFromStorage(OCP\Files\Storage\IStorage $sourceStorage, $sourceInternalPath, $targetInternalPath) at /var/www/owncloud/apps/ransomware_protection/lib/StorageWrapper.php#0

Comment (DeepDiver1975) [reply lib/public/Files/Storage.php]: The method signature differs ....

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by DeepDiver1975** [comment]
> php-cs-fixer is unhappy: https://drone.owncloud.com/owncloud/core/8952/69

aaaargh - I want the commit hook ....

**Comment by PVince81** [comment]
> aaaargh - I want the commit hook ....

you can have it: https://github.com/owncloud/core/pull/31837

**Comment by DeepDiver1975** [comment]
this is not yet fully working .... :fuck:

**Comment by phil-davis** [comment]
But getting better. Some ransomware acceptance tests pass, some get stufff like:
```
[Mon Jul 30 16:22:03 2018] Exception: {"Exception":"TypeError","Message":"Argument 1 passed to OCA\\Ransomware_Protection\\StorageWrapper::moveFromStorage() must be an instance of OCP\\Files\\Storage, instance of OCA\\Ransomware_Protection\\StorageWrapper given, called in \/home\/phil\/git\/owncloud\/core\/apps\/dav\/lib\/Connector\/Sabre\/File.php on line 243",
"Code":0,"Trace":"#0 \/home\/phil\/git\/owncloud\/core\/apps\/dav\/lib\/Connector\/Sabre\/File.php(243): OCA\\Ransomware_Protection\\StorageWrapper->moveFromStorage(Object(OCA\\Ransomware_Protection\\StorageWrapper), 'files\/testfile2...', 'files\/testfile2...')\n
#1 \/home\/phil\/git\/owncloud\/core\/apps\/dav\/lib\/Connector\/Sabre\/Directory.php(173): OCA\\DAV\\Connector\\Sabre\\File->put(Resource id #79)\n
#2 \/home\/phil\/git\/owncloud\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/Server.php(1095): OCA\\DAV\\Connector\\Sabre\\Directory->createFile('testfile2.txt', Resource id #79)\n
#3 \/home\/phil\/git\/owncloud\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/CorePlugin.php(525): Sabre\\DAV\\Server->createFile('testfile2.txt', Resource id #79, NULL)\n
#4 [internal function]: Sabre\\DAV\\CorePlugin->httpPut(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n
#5 \/home\/phil\/git\/owncloud\/core\/lib\/composer\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n
#6 \/home\/phil\/git\/owncloud\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/Server.php(479): Sabre\\Event\\EventEmitter->emit('method:PUT', Array)\n
#7 \/home\/phil\/git\/owncloud\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/Server.php(254): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n
#8 \/home\/phil\/git\/owncloud\/core\/apps\/dav\/appinfo\/v1\/webdav.php(64): Sabre\\DAV\\Server->exec()\n
#9 \/home\/phil\/git\/owncloud\/core\/remote.php(175): require_once('\/home\/phil\/git\/...')\n
#10 {main}","File":"\/home\/phil\/git\/owncloud\/core\/apps\/ransomware_protection\/lib\/StorageWrapper.php","Line":234}
```

**Comment by phil-davis** [comment]
Note: failures in ``update-testing`` are also related to this, e.g. https://drone.owncloud.com/owncloud/update-testing/97/64
```
An unhandled exception has been thrown:
TypeError: Argument 1 passed to OC\Avatar::__construct() must implement interface OCP\Files\Storage\IStorage, instance of OC\Files\Node\Folder given, called in /drone/src/owncloud/lib/private/Repair/MoveAvatarOutsideHome.php on line 112 and defined in /drone/src/owncloud/lib/private/Avatar.php:67
Stack trace:
#0 /drone/src/owncloud/lib/private/Repair/MoveAvatarOutsideHome.php(112): OC\Avatar->__construct(Object(OC\Files\Node\Folder), Object(OC\L10N\L10N), Object(OC\User\User), Object(OC\Log))
#1 /drone/src/owncloud/lib/private/Repair/MoveAvatarOutsideHome.php(135): OC\Repair\MoveAvatarOutsideHome->moveAvatars(Object(OC\Repair), Object(OC\User\User))
#2 /drone/src/owncloud/lib/private/User/Manager.php(438): OC\Repair\MoveAvatarOutsideHome->OC\Repair\{closure}(Object(OC\User\User))
#3 /drone/src/owncloud/lib/private/User/AccountMapper.php(235): OC\User\Manager->OC\User\{closure}(Object(OC\User\Account))
#4 /drone/src/owncloud/lib/private/User/Manager.php(439): OC\User\AccountMapper->callForAllUsers(Object(Closure), '', true)
#5 /drone/src/owncloud/lib/private/User/Manager.php(457): OC\User\Manager->callForAllUsers(Object(Closure), '', true)
#6 /drone/src/owncloud/lib/private/Repair/MoveAvatarOutsideHome.php(141): OC\User\Manager->callForSeenUsers(Object(Closure))
#7 /drone/src/owncloud/lib/private/Repair.php(91): OC\Repair\MoveAvatarOutsideHome->run(Object(OC\Repair))
#8 /drone/src/owncloud/lib/private/Updater.php(246): OC\Repair->run()
#9 /drone/src/owncloud/lib/private/Updater.php(108): OC\Updater->doUpgrade('10.1.0.1', '8.2.11.1')
#10 /drone/src/owncloud/core/Command/Upgrade.php(252): OC\Updater->upgrade()
#11 /drone/src/owncloud/lib/composer/symfony/console/Command/Command.php(251): OC\Core\Command\Upgrade->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /drone/src/owncloud/lib/composer/symfony/console/Application.php(946): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /drone/src/owncloud/lib/composer/symfony/console/Application.php(248): Symfony\Component\Console\Application->doRunCommand(Object(OC\Core\Command\Upgrade), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /drone/src/owncloud/lib/composer/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /drone/src/owncloud/lib/private/Console/Application.php(161): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /drone/src/owncloud/console.php(106): OC\Console\Application->run()
#17 /drone/src/owncloud/occ(11): require_once('/drone/src/ownc...')
```

**Comment by DeepDiver1975** [comment]
Only failing on smb ..... ready to be merged

@phil-davis can you please double check that this is working with the apps we had issues ... THX

**Comment by phil-davis** [comment]
I rebased this PR to make sure that the other ``Istorage`` related revert from yesterday in master is included here.

I can run sample firewall and ransomware_protection tests that fail without this core PR. They pass with this core PR - looks good.

Merge this, and then I will run the full app test suites on drone to confirm the world is great again.

**Pull request merged**